packages/task/src/browser/task-service.ts

```diff
@@ -454,6 +459,9 @@ export class TaskService implements TaskConfigurationClient {
      * It looks for configured and detected tasks.
      */
     async run(source: string, taskLabel: string, scope?: string): Promise<TaskInfo | undefined> {
+        // Open an empty terminal, display a message informing the user that terminal is starting
+        await this.taskTerminal.openEmptyTerminal(taskLabel);
```

Comment:
I see that code below can exit without reaching attach ever what should happen with the terminal then?

Reply:
@elaihau can a task be executed without the terminal? background tasks?

Reply:
> @elaihau can a task be executed without the terminal? background tasks?

@akosyakov sorry for replying so late.
as far as I can tell, we create terminalProcess for all tasks.

@marechal-p  please correct me if i am wrong